_includes/manuals/nightly/3.7_migration.md

```diff
@@ -0,0 +1,185 @@
+plugins. When a Foreman plugin is activated, database migration is perfored and
+new tables and indices are created. However when a plugin is uninstalled and
+deactivated, no tables are ever dropped and the data is left in the database
+forever. It's good idea to review at least OS installed package log to find out
+if any foreman package was ever uninstalled. Unfortunately, there is no list of
+tables for each plugin.
+
+#### Preparing the environment
+
+This blog post assumes that you already have the source and target databases
+set up. Make sure to setup the Postgresql database with UTF-8 enabled.
+
+The first step is to install required dependencies:
+
+    yum install -y foreman-postgresql
+
+Next, stop all your foreman instances and make sure, they stay stopped for
+good. On Red Hat compatible systems, you need to stop the `httpd` and (if
+applicable) `foreman-tasks` service. There is a helper utility called
+`foreman-maintain` when installed to ease the process of shutting down
+services. Make sure that neither Puppet nor a monitoring system automatically
+starts these services again.
+
+You can issue the MySQL command `SHOW FULL PROCESSLIST;` to show you all active
+connections to your MySQL database. Make sure, there are none.
+
+TODO: (WHY DO WE NEED THIS): If you are using the foreman-tasks plugin, you
+need to patch the script we are using to migrate the data. If you are not sure,
+wether you use this plugin, check if you have an entry `Tasks` in Foreman's
+Monitor menu. If you are still not sure, it is safe to apply the patch in any
+case.
+
+```sh
+wget -O /tmp/convert-tasks-patch.diff https://gist.githubusercontent.com/timogoebel/cd29e5206678ca0e0d78222af70583ec/raw/bbe998624098c1c3e656fe2582d33b0d9f9285aa/0001-convert-skip-foreman-tasks-tables.patch
+yum install patch -y
+cd /usr/share/foreman
+
+patch -p1 < /tmp/convert-tasks-patch.diff
+```
+
+The foreman user that has access to your Postgresql database and is used for
+the conversion temporarily needs super user privileges.
+
+```bash
+## postgres@postgres:~  $ su - postgres
+## postgres@postgres:~  $ psql
+psql (9.6.2)
+Type "help" for help.
+
+postgres=# ALTER USER foreman WITH SUPERUSER;
+ALTER ROLE
+postgres=#
+```
+
+#### Migrating
+
+Before proceeding migration, it's a good idea to permanently delete all audit
+records older than few days:
+
+    foreman-rake audits:expire days=10 RAILS_ENV="production"
+
+To permanently delete all reports older than 2 days:
+
+    foreman-rake reports:expire days=2 batch_size=50000 sleep_time=0.000001 RAILS_ENV="production"
+
+The command above deletes records in batches of 50k items, it can take a lot of
+time to process all records. In both cases, progress can be seen in the
+`production.log`.
+
+To migrate the data from one database to the other, we will use a rake task.
+You might already be familiar to this task from upgrading Foreman to a newer
+version.
+
+Command `foreman-rake -T` lists all available tasks. We will be using
```

Comment:
I think we can drop the -T sentence and combine the second one with the previous paragraph